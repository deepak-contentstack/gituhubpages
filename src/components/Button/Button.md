# Button

## Best practices

- Use clear and accurate labels.
- Start button labels with verbs, for example "Add field", not "New field".
- Keep the position of the buttons consistent on the interface.
- Reduce the complexity by using a small number of actions on the interface. It may create confusion if too many actions are used.
