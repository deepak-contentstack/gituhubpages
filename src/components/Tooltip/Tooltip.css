@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/variable.css';
@import '../../static/styles/css/common.css';

@mixin font-Inter {
    font-family: $font-family-primary;
}

.tippy-box[data-animation='fade'][data-state='hidden'] {
    opacity: $opacity-0;
}

.tippy-box[data-inertia][data-state='visible'] {
    transition-timing-function: cubic-bezier(0.54, 1.5, 0.38, 1.11);
}

.tippy-wrapper{
    display: inline-block;
}

.tippy-box-dark {
    background-color: $color-brand-secondary-dark;
    border-radius: $radii-4;
    color: $color-font-white;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: 0.04em;
    min-height:$space-22;
    transition-property: transform,visibility,opacity;
    @include font-Inter;
    
    >.tippy-content {
        line-height: $space-15;
        padding: $space-4 $space-8;
    }
}

.tippy-box-dark-tooltip {
    background-color: $color-brand-secondary-dark;
    border-radius: $radii-4;
    color: $color-font-white;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: 0.04em;
    min-height:$space-22;
    transition-property: transform,visibility,opacity;
    @include font-Inter;
    position: absolute;
    top: -30px;
    left: 0px;
    width:max-content;
    

    >.tippy-content {
        line-height: $space-15;
        padding: $space-4 $space-8;
    }
}

.tippy-box-light {
    background-color: $color-brand-white-base;
    border-radius: $radii-10;
    box-shadow: 0 4px 30px rgba(0, 0, 0, 0.25);
    color: $color-font-base;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: -6e-05em;
    text-align: left;
    @include font-Inter;

    &--primary {
        max-height: 22.81rem;
        max-width: 25rem!important;
        overflow: auto;
        padding: calc(var(--space-16) + 1px) $space-20;
        width: 25rem;

        > .tippy-content {
            .tippy-card {
                border-bottom: 1px solid $color-border-lighter;
                margin-bottom: $space-20;
                padding-bottom: $space-8;
                position: relative;

                .Env-label {
                    color: $color-font-base;
                    font-size: $size-font-small;
                    font-weight: $font-weight-medium;
                    margin-left: $space-5;
                }

                .tippy-card-content {
                    padding: $space-18 0 $space-18 $space-15;

                    .tippy-label {
                        color: $color-font-base;
                        font-size: $size-font-small;
                        font-weight: $font-weight-bold;
                    }
                    .tippy-value {
                        color:$color-font-tertiary;
                        font-size: $size-font-small;
                        font-weight: $font-weight-medium;
                        letter-spacing: -6e-5em;
                        margin-top: $space-5;
                    }
                    .tippy-lang {
                        color: $color-font-base;
                        font-size: $size-font-small;
                        font-weight: $font-weight-regular;
                    }
                }
                &:last-child {
                    border-bottom: 0;
                    margin-bottom: 0;
                    padding-bottom: 0;
                }
            }
        }
    }

    &--secondary {
        max-width: 22.5rem !important;
        padding: $space-15 $space-20;
        width: auto;

        >.tippy-content {
            line-height: $space-15;
            .tippy-content-title {
                color:$color-font-base;
                font-weight: $font-weight-bold;
                margin-bottom: $space-6;
            }
        }
    }


    &--dynamic {
        padding: $space-15 $space-20;
        > .tippy-content {
            line-height: $space-15;
            .tippy-content-title {
                color: $color-font-base;
                font-weight: $font-weight-bold;
                margin-bottom: $space-6;
            }
        }
    }
}

.TooltipWrapper {
    padding:  3.125rem 6.25rem;
}

.KeyShortcut {
    background: $color-brand-primary-base;
    border-radius: $space-5;
    color: $color-font-white;
    font-size: $size-font-tiny;
    font-weight: $font-weight-bold;
    height: $space-20;
    margin-left: $space-10;
    padding: $space-5 0;
    text-align: center;
    width: $space-20;
}

.tippy-box-light-menu {
    border-radius: $radii-6;
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
    color: $color-font-base;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    padding: $space-6 $space-10;
    position: relative;
    right: 1px;
    transition: 0ms !important;
    white-space: nowrap;
    .tippy-content {
        transition: 0ms !important;
    }
}

.tippy-arrow-light {
    border: $radii-8 solid transparent;
    height: 0;
    position: absolute;
    width: 0;
    
    &-left{
        align-self: center;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }

    &-left-start{
        align-self: flex-start;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }

    &-left-end{
        align-self: flex-end;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }
    
    &-right{
        align-self: center;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-right-start{
        align-self: flex-start;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-right-end{
        align-self: flex-end;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-top{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        left: calc(50% - var(--space-8));
        top: -$space-16;
        
    }

    &-top-start{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        left: calc(20% - var(--space-8));
        top: -$space-16;
    }

    &-top-end{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        right: calc(20% - var(--space-8));
        top: -$space-16;
    }
    
    &-bottom{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        left: calc(50% - var(--space-8));
    }

    &-bottom-start{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        left: calc(20% - var(--space-8));
    }

    &-bottom-end{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        right: calc(20% - var(--space-8));
    }
}

.tippy-arrow-dark {
    border: $radii-8 solid transparent;
    height: 0;
    position: absolute;
    width: 0;
    
    &-left{
        align-self: center;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }

    &-left-start{
        align-self: flex-start;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }

    &-left-end{
        align-self: flex-end;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }
    
    &-right{
        align-self: center;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-right-start{
        align-self: flex-start;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-right-end{
        align-self: flex-end;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-top{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        left: calc(50% - var(--space-8));
        top: -$space-16;
        
    }

    &-top-start{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        left: calc(20% - var(--space-8));
        top: -$space-16;
    }

    &-top-end{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        right: calc(20% - var(--space-8));
        top: -$space-16;
    }
    
    &-bottom{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        left: calc(50% - var(--space-8));
    }

    &-bottom-start{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        left: calc(20% - var(--space-8));
    }

    &-bottom-end{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        right: calc(20% - var(--space-8));
    }
}

.closeContainer{
    border-radius: $radii-50;
    cursor: pointer;
    height: $space-32;
    width: $space-32;
    &:hover{
        background: $color-brand-secondary-lighter;
        svg {
            path {
                stroke: $color-base-gray-20;
            }
        }
    }
}
.tippy-content {
    overflow-wrap: break-word;
}