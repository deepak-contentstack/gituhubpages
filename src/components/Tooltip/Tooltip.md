## **Tooltip**

A Tooltip Component can be used to indicate the user what is the functionality of certain element.

### **Tooltip Variant use**

- **dark**: Tooltip with dark variant can be used to show additional information for certain element.

  - props:
    - variant: dark

- **light**: Tooltip that has more details and components as a content (eg: Tooltip Card, Publish status.)

  - props:
    - variant: light

#### **Props**:

- **content**: String or React Component

  - required

- **position**: String

  - required
  - values : right | left | top | bottom

- **variantType**: String

  -optional (defaults to dark)
  -values : dark | light

- **showArrow**: Boolean

  - optional

- **trigger**: String

  - optional
  - default value : 'mouseenter focus'
  - other values: 'click', 'focusin', 'mouseenter click'

* **onHidden**: function
  Invoked once after tooltip hides.
  - optional


* **visible**: Boolean
  - Control the tooltips visibility
  - optional


* **showClose**: Boolean
  - Show a close icon at the top right corner of the tooltip
  - optional

* **onClose**: Functiom
  - Callback function to be executed after close icon is clicked
  - optional


