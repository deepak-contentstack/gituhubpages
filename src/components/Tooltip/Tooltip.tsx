import './Tooltip.css'

import React, { createRef } from 'react'

import cn from 'classnames'

import Tippy, { TippyProps } from '@tippyjs/react'

// import Icon from '../Icon2'

enum ArrowDirection {
  "left-start" = "right-start",
  "left-end" = "right-end",
  "left" = "right",

  "right-start" = "left-start",
  "right-end" = "left-end",
  "right" = "left",

  "top-start" = "bottom-start",
  "top-end" = "bottom-end",
  "top" = "bottom",

  "bottom-start" = "top-start",
  "bottom-end" = "top-end",
  "bottom" = "top",
};

enum PaddingClass {
  "left-start" = "pl-10",
  "left-end" = "pl-10",
  "left" = "pl-10",

  "right-start" = "pr-10",
  "right-end" = "pr-10",
  "right" = "pr-10",

  "top-start" = "pt-10",
  "top-end" = "pt-10",
  "top" = "pt-10",

  "bottom-start" = "pb-10",
  "bottom-end" = "pb-10",
  "bottom" = "pb-10",
};
interface ITooltip extends TippyProps {
  content: string | React.ReactNode | { text: string; icon: any }
  position: string
  showArrow?: boolean
  onCreate?: (data: any) => void
  trigger?: string
  onHidden?: () => any
  disabled?: boolean
  variantType?: "dark" | "light" | "menuIcon"
  visible?: boolean,
  showClose?: boolean,
  onClose?: () => void,
  type?: "primary" | "secondary" | "dynamic",
  maxWidth?: any,
  style?: React.CSSProperties,
  [propsKey: string]: any
  testId?: string
  customTooltipCss?: string,
  wrapperElementType?: "div" | "span",
}

const Tooltip: React.FunctionComponent<ITooltip> = (props: any) => {
  const node = createRef<HTMLInputElement>()
  const { content, position, children, isMenuIcon, visible, showClose, testId, style, customTooltipCss, ...rest } = props

  const showArrow = props.showArrow !== undefined ? props.showArrow : false
  const variant = props.variantType || "dark"
  const tooltipType = props.type !== undefined ? props.type : "primary"
  const disabled = props.disabled ? props.disabled : false
  const interactive = typeof content === "string" ? false : true;

  let className =
    variant === "dark"
      ? "tippy-box-dark"
      : tooltipType === "primary"
        ? "tippy-box-light tippy-box-light--primary"
        : `tippy-box-light tippy-box-light--${tooltipType}`
  className = variant === "menuIcon" ? "tippy-box-light tippy-box-light-menu" : className
  className = customTooltipCss ? customTooltipCss : className
  className = cn(className, props.className);
  const wrapperClassNames = cn("tippy-wrapper", { [`${PaddingClass[position]}`]: showArrow });

  const getContent = () => {
    if (variant === "menuIcon") {
      return getKeyIcon(content.text, content.icon);
    }

    if (showArrow) {
      return (
        <div className="flex">
          {content}
          <div className={`tippy-arrow-${variant} tippy-arrow-${variant}-${ArrowDirection[position]}`} />
        </div >
      );
    }

    return content;
  }

  const wrapWithClose = (content) => {
    return (
      <>
        <div className="flex-v-center">
          <div>{content}</div>
          <div className="closeContainer flex-v-center flex-h-center" onClick={props.onClose ? () => props.onClose() : null}>
            {/* <Icon icon='Cancel' /> */}
          </div>
        </div>

      </>
    );
  };

  const WrapperTag = getWrapperType(props.wrapperElementType || "div");
  return (
    <Tippy
      content={showClose ? wrapWithClose(getContent()) : getContent()}
      className={className}
      placement={position}
      arrow={showArrow}
      interactive={interactive}
      disabled={disabled}
      visible={visible}
      {...rest}
    >
      <WrapperTag ref={node} className={wrapperClassNames} data-test-id={testId} style={style}>
        {children}
      </WrapperTag>
    </Tippy>
  )
}

const getKeyIcon = (text: string, icon: any) => {
  return (
    <div className="flex-v-center">
      <div>{text}</div>
      <div className="KeyShortcut">{icon}</div>
    </div>
  )
}
const getWrapperType = (type: string) => {
  if (type === "span") {
    return "span";
  }
  return "div";
}

Tooltip.defaultProps = {
  testId: "cs-tooltip",
  wrapperElementType: "div",
  trigger: "mouseenter focusin"
}

export default Tooltip
