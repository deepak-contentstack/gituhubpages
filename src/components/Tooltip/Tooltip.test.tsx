import React from 'react'
import { mount } from 'enzyme';
import { render, cleanup } from '@testing-library/react'
import 'regenerator-runtime/runtime'

import Tooltip from './Tooltip'
// import Icon from '../Icon/Icon'
import Button from '../Button/Button';

afterEach(cleanup)

//The Unit testing for all cases has been covered using @testing-library/react
//The snapshot testing is not covered as there was some issue due to use of third party package.

describe('Tooltip', () => {
  let instance: any = null

  afterEach(() => {
    instance = null
  })

  it('renders only the child element', () => {
    const stringContent = render(
      <Tooltip content="tooltip" position="right">
        <button />
      </Tooltip>
    )
    expect(stringContent.container.innerHTML).toBe(
      '<div class=\"tippy-wrapper\" data-test-id=\"cs-tooltip\"><button></button></div>'
    )
    expect(stringContent.container.innerHTML).toMatchSnapshot()
  })

  it('adds a tippy instance to the child node', () => {
    render(
      <Tooltip content={<strong>tooltip</strong>} position="right" onCreate={i => (instance = i)}>
        <button>Click me!</button>
      </Tooltip>
    )
    expect(instance).not.toBeNull()
  })

  it('renders react element content inside the content prop', () => {
    render(
      <Tooltip content={<strong>tooltip</strong>} position="right" onCreate={i => (instance = i)}>
        <button>Click me!</button>
      </Tooltip>
    )

    expect(instance.popper.querySelector('strong')).not.toBeNull()
  })

  it('unmount destroys the tippy instance', () => {
    const { container, unmount } = render(
      <Tooltip content="tooltip" position="right" onCreate={i => (instance = i)}>
        <button />
      </Tooltip>
    )
    const button: any = container.querySelector('button')

    unmount()

    expect(button._tippy).toBeUndefined()
    expect(instance.state.isDestroyed).toBe(true)
  })

  it('component as a child', () => {
    const Child = () => {
      return <div></div>
    }

    render(
      <Tooltip content="tooltip" position="right" onCreate={i => (instance = i)}>
        <Child />
      </Tooltip>
    )
    expect(instance).not.toBeNull()
  })

  it('Render Tooltip with arrow', () => {
    const TooltipComponent = () => {
      return <div>
        <Tooltip
          content="Plain text to show additional information."
          position="bottom"
          showArrow={true}
          variantType="light"
          type="primary"
          visible={true}
        ></Tooltip>
      </div>
    }
    const tree = render(<TooltipComponent />)
    expect(tree).toMatchSnapshot()
  })

  it('Render Tooltip with Close', () => {
    const TooltipComponent = () => {
      return <div>
        <Tooltip
          content="Plain text to show additional information."
          position="bottom"
          showArrow={true}
          variantType="light"
          type="primary"
          visible={true}
          showClose={true}
        ></Tooltip>
      </div>
    }
    const tree = render(<TooltipComponent />)
    expect(tree).toMatchSnapshot()
  })

  it('onClick of Tooltip with Close', () => {
    const mockAction = jest.fn();
    const TooltipComponent = () => {
      return <div>
        <Tooltip
          content="Plain text to show additional information."
          position="bottom"
          showArrow={true}
          variantType="light"
          type="primary"
          visible={true}
          showClose={true}
          onClose={() => { mockAction() }}
        >
          <Button buttonType="primary">Open tooltip</Button>
        </Tooltip>
      </div>
    }
    const tree = mount(<TooltipComponent />)
    tree.find('.closeContainer').simulate('click')
    expect(mockAction).toHaveBeenCalledTimes(1);
  })

  it('Render MenuIcon type Tooltip', () => {
    const TooltipComponent = () => {
      return <div>
        <div className="TooltipWrapper">
          <Tooltip content={{ text: 'abc', icon: 'A' }} position="right" variantType="menuIcon"></Tooltip>
        </div >
      </div>
    }
    const tree = render(<TooltipComponent />)
    expect(tree).toMatchSnapshot()
  })
})

// describe('Tooltip', () => {

//     it('render Tooltip with string as Content', () => {
//         const DefaultTooltip = () => {
//             return (
//                 <Tooltip content="String as Content" position="right">
//                     <Icon icon="Logo" size="small" />
//                 </Tooltip>
//             )
//         }
//         const Dummy = () => {
//             return (
//                 <h1>hello</h1>
//             )
//         }
//         const tree = renderer.create(<Dummy />);
//         console.log('tree....', tree.toJSON());

//         expect(tree).toMatchSnapshot();
//     })

//     it('render Tooltip with Component as Content', () => {
//         const CustomComponentAsContent = () => {
//             const CustomComponent = () => {
//                 return (
//                     <div>
//                         <span>Custom Component Data</span>
//                     </div>
//                 )
//             }
//             return (
//                 <Tooltip content={<CustomComponent />} position="right">
//                     <Button buttonType='primary'>Hover on me!</Button>
//                 </Tooltip>
//             )
//         }
//         const tree = renderer.create(<CustomComponentAsContent />).toJSON();
//         expect(tree).toMatchSnapshot();
//     })

//     it('render Tooltip with Component as Children', () => {
//         const CustomComponentAsChildren = () => {
//             const CustomComponent = () => {
//                 return (
//                     <div>
//                         <span>Custom Component Children</span>
//                     </div>
//                 )
//             }
//             return (
//                 <Tooltip content='Children is Custom component' position="right">
//                     <CustomComponent />
//                 </Tooltip>
//             )
//         }
//         const tree = renderer.create(<CustomComponentAsChildren />).toJSON();
//         expect(tree).toMatchSnapshot();
//     })
// })
