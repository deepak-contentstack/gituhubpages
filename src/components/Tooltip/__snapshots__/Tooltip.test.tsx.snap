// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Render MenuIcon type Tooltip 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      data-tippy-root=""
      id="tippy-8"
      style="pointer-events: none; z-index: 9999; visibility: visible; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 10px);"
    >
      <div
        class="tippy-box tippy-box-light tippy-box-light--primary"
        data-animation="fade"
        data-escaped=""
        data-placement="bottom"
        data-reference-hidden=""
        data-state="visible"
        role="tooltip"
        style="max-width: 350px; transition-duration: 300ms;"
        tabindex="-1"
      >
        <div
          class="tippy-content"
          data-state="visible"
          style="transition-duration: 300ms;"
        >
          <div>
            <div
              class="flex-v-center"
            >
              <div>
                <div
                  class="flex"
                >
                  Plain text to show additional information.
                  <div
                    class="tippy-arrow-light tippy-arrow-light-top"
                  />
                </div>
              </div>
              <div
                class="closeContainer flex-v-center flex-h-center"
              />
            </div>
          </div>
        </div>
        <div
          class="tippy-arrow"
          style="position: absolute; left: 0px; transform: translate(3px, 0px);"
        />
      </div>
    </div>
    <div>
      <div>
        <div
          class="TooltipWrapper"
        >
          <div
            aria-expanded="false"
            class="tippy-wrapper"
            data-test-id="cs-tooltip"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="TooltipWrapper"
      >
        <div
          aria-expanded="false"
          class="tippy-wrapper"
          data-test-id="cs-tooltip"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tooltip Render Tooltip with Close 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="tippy-wrapper pb-10"
          data-test-id="cs-tooltip"
        />
      </div>
    </div>
    <div
      data-tippy-root=""
      id="tippy-7"
      style="pointer-events: none; z-index: 9999; visibility: visible; transition: none; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 10px);"
    >
      <div
        class="tippy-box tippy-box-light tippy-box-light--primary"
        data-animation="fade"
        data-escaped=""
        data-placement="bottom"
        data-reference-hidden=""
        data-state="hidden"
        role="tooltip"
        style="max-width: 350px; transition-duration: 0ms;"
        tabindex="-1"
      >
        <div
          class="tippy-content"
          data-state="hidden"
          style="transition-duration: 0ms;"
        >
          <div>
            <div
              class="flex-v-center"
            >
              <div>
                <div
                  class="flex"
                >
                  Plain text to show additional information.
                  <div
                    class="tippy-arrow-light tippy-arrow-light-top"
                  />
                </div>
              </div>
              <div
                class="closeContainer flex-v-center flex-h-center"
              />
            </div>
          </div>
        </div>
        <div
          class="tippy-arrow"
          style="position: absolute; left: 0px; transform: translate(3px, 0px);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="tippy-wrapper pb-10"
        data-test-id="cs-tooltip"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tooltip Render Tooltip with arrow 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="tippy-wrapper pb-10"
          data-test-id="cs-tooltip"
        />
      </div>
    </div>
    <div
      data-tippy-root=""
      id="tippy-6"
      style="pointer-events: none; z-index: 9999; visibility: visible; transition: none; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 10px);"
    >
      <div
        class="tippy-box tippy-box-light tippy-box-light--primary"
        data-animation="fade"
        data-escaped=""
        data-placement="bottom"
        data-reference-hidden=""
        data-state="hidden"
        role="tooltip"
        style="max-width: 350px; transition-duration: 0ms;"
        tabindex="-1"
      >
        <div
          class="tippy-content"
          data-state="hidden"
          style="transition-duration: 0ms;"
        >
          <div>
            <div
              class="flex"
            >
              Plain text to show additional information.
              <div
                class="tippy-arrow-light tippy-arrow-light-top"
              />
            </div>
          </div>
        </div>
        <div
          class="tippy-arrow"
          style="position: absolute; left: 0px; transform: translate(3px, 0px);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="tippy-wrapper pb-10"
        data-test-id="cs-tooltip"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tooltip renders only the child element 1`] = `"<div class=\\"tippy-wrapper\\" data-test-id=\\"cs-tooltip\\"><button></button></div>"`;
